sdk/appconfiguration/azure-mgmt-appconfiguration/azure/mgmt/appconfiguration/_app_configuration_management_client.py

```diff
@@ -70,11 +73,18 @@ def __init__(
         self,
         credential: "TokenCredential",
         subscription_id: str,
-        base_url: str = "https://management.azure.com",
+        base_url: str = "",
```

Comment:
We can't remove default value since `base_url` without default value is changed from optional to required which is breakings for existing SDK users.

Reply:
i think the default has been changed to empty string and will be set according to cloud setting.

Reply:
Yes. Both of you are correct.

We cannot change the base_url from optional to required. That's definitely a breaking change.

What we want to achieve is to be able to differentiate between user explicitly provided the base_url or they just used the default value.

If base_url is explicitly provided, we need to honor it. If only default value is used, we can get it from cloud_setting.